agent/api/eni/eni.go

```diff
@@ -171,5 +197,15 @@ func ValidateTaskENI(acsenis []*ecsacs.ElasticNetworkInterface) error {
 		return errors.Errorf("eni message validation: empty eni id in the message")
 	}
 
+	if (aws.StringValue(acsenis[0].InterfaceType) != BranchENIName) && (aws.StringValue(acsenis[0].InterfaceType) != RegularENIName) {
+		return errors.Errorf("Invalid ENI interface type: %s", aws.StringValue(acsenis[0].InterfaceType))
+	}
+
+	if aws.StringValue(acsenis[0].InterfaceType) == BranchENIName {
+		if acsenis[0].InterfaceVlanProperties == nil {
+			return errors.Errorf("vlan interface properties missing.")
```

Comment:
can you also check that vlan id and trunk mac address fields are not empty here?